### PR TITLE
#INC-1030 - Stripe call timeout

### PR DESCRIPTION
**Pull request by d12frosted**
@artempWix please take this PR and add tests ❤️

### CODE REVIEW

**Review by dleshem — CHANGES_REQUESTED** [review]
timeouts should be passed in constructor.

**Comment by artempWix** [comment]
@dleshem, optional constructor parameters were added
now these changes should affect only Cashier (we will pass these timeouts)

**Comment by dleshem** [comment]
@artempWix Excellent, thanks.

Consider making these arguments `Option[Duration]` instead of `Option[Int]`. You can then pass `Some(2.seconds)` instead of `Some(1000 * 2)` which makes the code more readable.

File: libpay-stripe/src/main/scala/com/wix/pay/stripe/StripeGateway.scala [hunk 1]
```diff
@@ -48,8 +52,17 @@ class StripeGateway(merchantParser: StripeMerchantParser = new JsonStripeMerchan
...
 
-  private def requestOptionsFor(apiKey: String): RequestOptions = {
-    RequestOptions.builder.setApiKey(apiKey).build
+  private def requestOptionsFor(apiKey: String): RequestOptions = RequestOptions.builder
+    .setApiKey(apiKey)
+    .setWith(_.setConnectTimeout, connectTimeout)
+    .setWith(_.setReadTimeout, readTimeout)
+    .build
+
+  private implicit class `RequestOptionsTimeoutBuilder`(builder: RequestOptionsBuilder) {
```
Comment (d12frosted) [inline libpay-stripe/src/main/scala/com/wix/pay/stripe/StripeGateway.scala]: `RequestOptionsTimeoutBuilder`, `maybeTimeout` and `timeout` should be renamed I think :)

**Review by dleshem — APPROVED** [review]

**Pull request merged**